app/src/main/java/com/example/calculator/Calculation.java

```diff
@@ -5,13 +5,57 @@
 public class Calculation {
 
     private ArrayList<Token> mStringArrayList;
+    private double answer;
+    private boolean errorOccurred;
+    private String errorMessage;
 
     public Calculation(ArrayList<Character> arr) {
-         mStringArrayList = setTokenArrayList(arr);
+        errorOccurred = false;
+        errorMessage = "No Error";
+        mStringArrayList = setTokenArrayList(arr);
+        pairParenthesis();
+        answer = solveAnswer();
     }
 
-    public double solveAnswer() {
+    public double getAnswer() {
+        return answer;
+    }
+
+    public boolean isErrorOccurred() {
+        return errorOccurred;
+    }
+
+    public String getErrorMessage() {
+        return errorMessage;
+    }
+
+    private void pairParenthesis() {
+        int opening = 0;
+        int closing = 0;
+        for(Token t : mStringArrayList) {
+            if(t instanceof Parenthesis) {
+                if(t.getOperator() == '(') {
+                    opening++;
+                } else {
+                    closing++;
+                }
+            }
+        }
+        if(opening != closing) {
+            int addParenthesis = opening - closing;
+            for(int i = 0; i < Math.abs(addParenthesis); i++) {
```

Comment:
Math.abs() will be exeucted whenever it checks the terminal condition.
How can you avoid it?